bot/cogs/bot.py

```diff
@@ -100,81 +100,220 @@ def __init__(self, bot: AutoShardedBot):
         embed = Embed(description=text)
         await ctx.send(embed=embed)
 
-    def codeblock_stripping(self, msg: str):
+    def codeblock_stripping(self, msg: str, bad_ticks: bool):
         """
         Strip msg in order to find Python code.
 
         Tries to strip out Python code out of msg and returns the stripped block or
         None if the block is a valid Python codeblock.
         """
         if msg.count("\n") >= 3:
-            # Filtering valid Python codeblocks and exiting if a valid Python codeblock is found
-            if re.search("```(?:py|python)\n(.*?)```", msg, re.IGNORECASE | re.DOTALL):
+            # Filtering valid Python codeblocks and exiting if a valid Python codeblock is found.
+            if re.search("```(?:py|python)\n(.*?)```", msg, re.IGNORECASE | re.DOTALL) and not bad_ticks:
                 log.trace("Someone wrote a message that was already a "
                           "valid Python syntax highlighted code block. No action taken.")
-                return None
+                return None, None
+
             else:
                 # Stripping backticks from every line of the message.
                 log.trace(f"Stripping backticks from message.\n\n{msg}\n\n")
                 content = ""
-                for line in msg.splitlines():
-                    content += line.strip("`") + "\n"
+                for line in msg.splitlines(keepends=True):
+                    content += line.strip("`")
 
                 content = content.strip()
 
-                # Remove "Python" or "Py" from top of the message if exists
+                # Remove "Python" or "Py" from top of the message if exists.
                 log.trace(f"Removing 'py' or 'python' from message.\n\n{content}\n\n")
+                pycode = False
                 if content.lower().startswith("python"):
                     content = content[6:]
+                    pycode = True
                 elif content.lower().startswith("py"):
                     content = content[2:]
+                    pycode = True
+
+                if pycode:
+                    content = content.splitlines(keepends=True)
+
+                    # Check if there might be code in the first line, and preserve it.
+                    first_line = content[0]
+                    if " " in content[0]:
+                        first_space = first_line.index(" ")
+                        content[0] = first_line[first_space:]
+                        content = "".join(content)
+
+                    # If there's no code we can just get rid of the first line.
+                    else:
+                        content = "".join(content[1:])
 
                 # Strip again to remove the whitespace(s) left before the code
```

Comment:
this block comment is a bit poor, here's a suggestion for improvement:

```py
# Strip it again to remove any leading whitespace. This is neccessary
# if the first line of the message looked like ```python <code>
```